### ISSUE TITLE
IRSA IAM Role and Launch template name mismatch - Provisioning Error

### ISSUE THREAD

**Issue by vazkarvishal** [terraform-aws-modules/terraform-aws-eks#2431]
## Description

When using the module without any modifications, it creates new IAM Roles for IRSA. This IRSA role has a very specific property which restricts the `"Action": "ec2:RunInstances"` operation to resource names starting with `launch-template/*`. When using Karpenter with NodeTemplates, we do not control what name the Launch template gets and thus, by default the Launch Template acquires a name which matches that of the installation/cluster (for e.g. **projectXYZ-123**). 

Due to this reason, new node provisioning fails with a `Client.UnauthorizedOperation` error for the `RunInstances` API call. 

How can we change this? so that the Resource (launch template) names starting with cluster defaults will not result in Access Denied issues?

## Versions

- Module version [Required]: 19.5.1

- Terraform version: 1.1.3
- Provider version(s):
AWS: 4.0

## Reproduction Code [Required]

- Provision the Karpenter controller via the module
- Create a provisioner with the AWSNodeTemplate resource 
- Use the node template in the Provisioner
- Enable Karpenter logs in debug mode
- Use CloudTrail to view failure logs

## Expected behavior

Nodes should provision without an error. 

## Actual behavior

Permissions too strict and unable to amend.

### Terminal Output Screenshot(s)

![image](https://user-images.githubusercontent.com/13076513/215542467-7bed690b-493e-483b-ba73-9f4163b585f4.png)
![image](https://user-images.githubusercontent.com/13076513/215543219-c4239134-02e4-48d6-b96e-a4ef20a2d680.png)

**Comment by bryantbiggs** [comment]
1. Have you seen our examples
2. Please fill in the requested details in the issue form - without a reproduction we cannot offer any support https://github.com/bryantbiggs/how-to-create-reproduction

**Comment by vazkarvishal** [comment]
Thanks @bryantbiggs. I figured out what it was eventually. This was happening because the IAM policy which gets created out of the box by this module strictly locks down all API operations for `RunInstances` to the same account. 

In my case, the subnets and networks were shared using RAM and thus the policy was not allowing Karpenter to provision nodes in subnets from another account. I created another policy and attached it to the IAM Role which has the necessary permissions, and it seems to work now. Will close the issue and leave this comment on for reference, in case anyone else faces the same face palm moment.
